apps/src/javalab/JavabuilderConnection.js

```diff
@@ -20,7 +21,8 @@ export default class JavabuilderConnection {
       data: {
         projectUrl: dashboard.project.getProjectSourcesUrl(),
         channelId: this.channelId,
-        projectVersion: dashboard.project.getCurrentSourceVersionId()
+        projectVersion: dashboard.project.getCurrentSourceVersionId(),
+        levelId: this.levelId
```

Comment:
Will we need the full level URL? Thinking since we're using the same Javabuilder environment across all our dashboard environments, we'll likely want to disambiguate which environment to make the request to. I'm not sure how our level IDs get set, but they don't seem to be consistent between environments.

Reply:
We provide the dashboard hostname in the session token (`iss` field). We are going to be making requests to dashboard to get level info so I think that's all we will need.

Reply:
[Changing to non-blocking] Do we need to be concerned at all about the route changing? i.e. if there's a similar change to Dani's stage -> lesson change, how can we ensure Javabuilder doesn't break?

Reply:
can you clarify what you mean by that? We will be accessing the route added [here](https://github.com/code-dot-org/code-dot-org/pull/40707). If we end up changing that specific route we would hit issues, but I am not sure why that would occur. If we wanted to send that specific route, we would also then need to add the validation route to the token in the future, instead of using the level id for both.

Reply:
Ok. That makes sense. We ended up sending the full url for projectSources because our projects backend is somewhat likely to change in the future and because it's rather complicated to build the url. Levels is much more stable and the route is clearly defined. This does feel like a good spot for an integration test (when we get a framework set up for that in the future) I added a task here so we can keep that in mind: https://codedotorg.atlassian.net/browse/CSA-363